torch/nn/quantized/dynamic/modules/linear.py

```diff
@@ -62,7 +62,12 @@ def from_float(cls, mod):
         """
         assert type(mod) == NNLinear, 'nn.quantized.dynamic.Linear.from_float only works for nn.Linear'
         assert hasattr(mod, 'qconfig'), 'Input float module must have qconfig defined'
-        weight_observer = mod.qconfig.weight()
+        if mod.qconfig is not None and mod.qconfig.weight() is not None:
+            weight_observer = mod.qconfig.weight()
+        else:
+            from torch.quantization.observer import default_weight_observer
```

Comment:
This works, but for clarity, please move the import to the beginning of the file and do:
```
   
def from_float(cls, mod):
...
        if mod.qconfig is not None and mod.qconfig.weight() is not None:
            weight_observer = mod.qconfig.weight()
        else:
            weight_observer = default_weight_observer
...
```

Reply:
The problem is the circular import issues in Python: https://github.com/pytorch/pytorch/pull/23128#discussion_r310745665. The current solution is to postpone the import of `torch.quantization.observer` until you need it: https://stackoverflow.com/a/10027101/3234205

Reply:
The ideal solution is to move the observer and qconfig into a separate submodule (https://github.com/pytorch/pytorch/pull/23128#discussion_r311383406). However, it requires a new namespace, which involves many changes. I had discussions with @jerryzh168 and @zafartahirov for this issue.

Reply:
This is a valid workaround for the circular dependencies. I think we should add a comment though -- in case some future eng decides to move it out

Reply:
Added a comment here.